apps/judicial-system/web/src/routes/Prosecutor/Indictments/Indictment/Offenses/Offenses.tsx

```diff
@@ -0,0 +1,321 @@
+import { Dispatch, SetStateAction, useMemo, useState } from 'react'
+import InputMask from 'react-input-mask'
+import { useIntl } from 'react-intl'
+
+import { Box, Icon, Input, Select, Tag } from '@island.is/island-ui/core'
+import {
+  IndictmentCountOffense,
+  SubstanceMap,
+} from '@island.is/judicial-system/types'
+import { SectionHeading } from '@island.is/judicial-system-web/src/components'
+import {
+  Case,
+  Offense,
+} from '@island.is/judicial-system-web/src/graphql/schema'
+import { TempIndictmentCount } from '@island.is/judicial-system-web/src/types'
+import { isNonEmptyArray } from '@island.is/judicial-system-web/src/utils/arrayHelpers'
+import {
+  removeErrorMessageIfValid,
+  validateAndSetErrorMessage,
+} from '@island.is/judicial-system-web/src/utils/formHelper'
+import { UpdateIndictmentCount } from '@island.is/judicial-system-web/src/utils/hooks'
+import useOffenses from '@island.is/judicial-system-web/src/utils/hooks/useOffenses'
+
+import { Substances } from '../Substances/Substances'
+import { SpeedingOffenseFields } from './SpeedingOffenseFields'
+import { indictmentCount as strings } from '../IndictmentCount.strings'
+import { indictmentCountEnum as enumStrings } from '../IndictmentCountEnum.strings'
+
+const sortByCreatedDate = (o1: Offense, o2: Offense) => {
+  if (!(o1.created && o2.created)) {
+    return 0
+  }
+  if (o1.created < o2.created) {
+    return -1
+  }
+  if (o1.created > o2.created) {
+    return 1
+  }
+  return 0
+}
+
+const getUpdatedOffenses = (
+  currentOffenses: Offense[],
+  updatedOffense: Offense,
+) =>
+  [
+    ...currentOffenses.filter((o) => o.id !== updatedOffense.id),
+    updatedOffense,
+  ].sort(sortByCreatedDate)
```

Comment:
Before, it was sorted in the same order as the offenses appear in the dropdown list. But here we maintain the order they are selected which imo makes more sense as it is the "order" that the user selects.